### ISSUE TITLE
beam slope

### ISSUE THREAD

**Issue by rettinghaus** [music-encoding/music-encoding#648]
The `@slope` attribute on `<beam>` currently allows decimal values, although the schema doesn't say anything about the meaning of this. Do the allowed values make sense at all? I guess 
`data.DEGREES` would be better here.

**Comment by craigsapp** [comment]
Yes, fractional values would be required, since the endpoints of a slur are typically locked to specific positions in relation to the staff lines (they are usually tangent to the staff line, or centered on a staff line). In order to do this, the angle needs to be a continuous floating-point number.

The `@slope` attribute is not very useful and should be calculated as a byproduct of the stem lengths of the notes attached at the endpoints.  Here is a demo:

<img width="1011" alt="Screen Shot 2020-02-25 at 08 01 40" src="https://user-images.githubusercontent.com/3487289/75265542-e9aef880-57a5-11ea-8fcc-ca5e982b36b8.png">

Notice that the stem lengths of the two notes remain fixed at one octave in length, but the slope of the beam between them changes based on the distance between the notes.  The only particularly interesting value is 0.0 to indicate a horizontal beam, where the distance between the notes is no longer fixed.   There will be cases where the slope is too large because the notes are too close together, but to reduce the slope, the stem lengths should be adjusted: either one stem should be lengthened, or another should be shortened.  In other words, if you provide the slope, then also one stem length would also need to be provided in order to calculate the stem length of the notes at the other endpoint.

Another thought: The endpoints of the beams are (usually) quantized to specific vertical positions.  If you use a beam slope to set these positions, the endpoints will be approximately at the quantized positions, but not exactly.   In other words, there will be rounding error in converting the slope to vertical positions and any change in the horizontal layout would also mess up alignment with these quantized vertical positions.

So the only useful application of @slope would be when one of the endpoints of the beam does not attach to a note.


<img width="683" alt="Screen Shot 2020-02-25 at 08 30 04" src="https://user-images.githubusercontent.com/3487289/75268430-575e2300-57ac-11ea-8c74-e9fe4f1cc3f1.png">

**Comment by rettinghaus** [comment]
If we could refine `@slope` and extract it into a specific attribute class (e.g. `att.sloped`) we could use this in multiple places like hairpins (see #733) and slurs. 

What about something like 

> Defines the slope of the object by specifying the relative vertical distance of its start and end points in virtual units. 

This would allow to encode unusual steep beams or give better impression of the slur design. 
![image](https://user-images.githubusercontent.com/7693447/159663608-b0162d8e-5a99-443b-b61d-e655b558beee.png)
![image](https://user-images.githubusercontent.com/7693447/159663636-eb4bb7e8-0d1f-494f-ba32-fbceb85fde8a.png)

`@beam.slope` should also be improved to hold the maximal allow value instead of a static one, since you definitely don't want _all_ beams sloped, but maybe want all beams to be flat?
